models/model.js

```diff
@@ -32,4 +30,69 @@ exports.selectArticleById = (id) => {
     .then((results) => {
         return results.rows
     })
-}
\ No newline at end of file
+}
+exports.selectArticles = (
+  sortBy = "created_at",
+  orderBy = "DESC",
+  topic,
+) => {
+  let topicCondition = false;
+  if (!topic) {
+    topicCondition = true;
+  }
+  const allowedSortBy = [
+    "article_id",
+    "title",
+    "body",
+    "votes",
+    "author",
+    "topic",
+    "created_at",
+    "comment_count",
+  ];
+
+  if (!allowedSortBy.includes(sortBy)) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+
+  const allowedOrder = ["ASC", "DESC", "asc", "desc"];
+  if (!allowedOrder.includes(orderBy)) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+  return db.query(
+      `SELECT articles.*, COUNT(comments.comment_id) AS comment_count
+      FROM articles
+      LEFT JOIN comments
+      ON articles.article_id = comments.article_id
+      WHERE ${topicCondition} OR topic = $1

```

Comment:
This is going to leave you open to SQL injection. We should not be  inserting directly into our SQL query. I am also unsure as to why we are inserting a boolean into the SQL query? We can just make the where constraint conditional based upon whether we have been passed a topic.